keras_nlp/models/roberta/roberta_preprocessing.py

```diff
@@ -0,0 +1,140 @@
+
+    Input should be either a `tf.RaggedTensor` or a dense `tf.Tensor`, and
+    either rank-1 or rank-2.
+
+    Please refer to the arguments of `keras_nlp.layers.MultiSegmentPacker` for
+    more details.
+    """
+
+    def __init__(
+        self,
+        sequence_length,
+        start_value,
+        end_value,
+        pad_value=None,
+        truncate="round_robin",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.sequence_length = sequence_length
+        if truncate not in ("round_robin", "waterfall"):
+            raise ValueError(
+                "Only 'round_robin' and 'waterfall' algorithms are "
+                "supported. Received %s" % truncate
+            )
+        self.truncate = truncate
+        self.start_value = start_value
+        self.end_value = end_value
+        self.pad_value = pad_value
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "sequence_length": self.sequence_length,
+                "start_value": self.start_value,
+                "end_value": self.end_value,
+                "pad_value": self.pad_value,
+                "truncate": self.truncate,
+            }
+        )
+        return config
+
+    def _trim_inputs(self, inputs):
+        """Trim inputs to desired length."""
+        # Special tokens include the start token at the beginning of the
+        # sequence, two `end_value` at the end of every segment save the last,
+        # and the `end_value` at the end of the sequence.
+        num_special_tokens = 2 * len(inputs)
+        if self.truncate == "round_robin":
+            return tf_text.RoundRobinTrimmer(
+                self.sequence_length - num_special_tokens
+            ).trim(inputs)
+        elif self.truncate == "waterfall":
+            return tf_text.WaterfallTrimmer(
+                self.sequence_length - num_special_tokens
+            ).trim(inputs)
+        else:
+            raise ValueError("Unsupported truncate: %s" % self.truncate)
+
+    def _combine_inputs(self, segments):
+        """Combine inputs with start and end values added."""
+        dtype = segments[0].dtype
+        batch_size = segments[0].nrows()
+
+        start_value = tf.convert_to_tensor(self.start_value, dtype=dtype)
+        end_value = tf.convert_to_tensor(self.end_value, dtype=dtype)
+
+        start_column = tf.fill((batch_size, 1), start_value)
+        end_column = tf.fill((batch_size, 1), end_value)
+
+        segments_to_combine = []
+        for i, seg in enumerate(segments):
+            segments_to_combine.append(start_column if i == 0 else end_column)
+            segments_to_combine.append(seg)
+            segments_to_combine.append(end_column)
+
+        token_ids = tf.concat(segments_to_combine, 1)
+        return token_ids
+
+    def call(self, inputs):
+        if not isinstance(inputs, (list, tuple)):
```

Comment:
i don't think we need this, because we aren't exposing this standalone, and the preprocessor layer has the same check